### PR TITLE
Move statCache struct to internal/cache/metadata

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Did sanity testing locally, mounting, listing, file and directory reads etc.
2. Unit tests - All rand and passed locally. No new tests passed.
3. Integration tests - Tested through presubmit.

### CODE REVIEW

**Comment by gargnitingoogle** [comment]
@sethiay @raj-prince  Surprizingly the presubmit integration tests for this PR are not running despite setting the label. Log: `No need to execute tests` . 

This PR has code change, and I expected that this PR will have the test run because it's being merged to a *_release branch, but no. Any idea ?

**Comment by sethiay** [comment]
> @sethiay @raj-prince Surprizingly the presubmit integration tests for this PR are not running despite setting the label. [Log](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/b7ff859a-79b1-49a8-a2f0-62c5a310603f/log): `No need to execute tests` .
> 
> This PR has code change, and I expected that this PR will have the test run because it's being merged to a *_release branch, but no. Any idea ?

@Tulsishah can you please look into this ?

**Comment by Tulsishah** [comment]
> > @sethiay @raj-prince Surprizingly the presubmit integration tests for this PR are not running despite setting the label. [Log](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/b7ff859a-79b1-49a8-a2f0-62c5a310603f/log): `No need to execute tests` .
> > This PR has code change, and I expected that this PR will have the test run because it's being merged to a *_release branch, but no. Any idea ?
> 
> @Tulsishah can you please look into this ?

It rebuild it. It's in progress now.

**Comment by Tulsishah** [comment]
> @sethiay @raj-prince Surprizingly the presubmit integration tests for this PR are not running despite setting the label. [Log](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/b7ff859a-79b1-49a8-a2f0-62c5a310603f/log): `No need to execute tests` .
> 
> This PR has code change, and I expected that this PR will have the test run because it's being merged to a *_release branch, but no. Any idea ?

It looks like you push the commit, kokoro triggered and after that you attached the label.

**Comment by gargnitingoogle** [comment]
Integration tests passed. [Log](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/4bd8889e-a133-4c53-821a-1bdcb8f66fa8/log)

File: internal/cache/metadata/test/stat_cache_test.go [hunk 1]
```diff
@@ -12,13 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package caching_test
+package test
```
Comment (raj-prince) [inline internal/cache/metadata/test/stat_cache_test.go:15]: Why are moving this inside test package?

Generally we keep the go file and their unit test on the same level.

**Review by raj-prince — COMMENTED** [review]
Overall looks good.
Left a comment.

File: internal/cache/metadata/test/stat_cache_test.go (hunk 1 above)
Comment (gargnitingoogle) [reply internal/cache/metadata/test/stat_cache_test.go:15]: @raj-prince  This looks like old code. I'll see what went wrong. Thanks for pointing out..

Comment (gargnitingoogle) [reply internal/cache/metadata/test/stat_cache_test.go]: Found the issue. Pushed the updated code, which uses package metadata_test, as it should.  Marking this resolved.

**Comment by gargnitingoogle** [comment]
> Overall looks good. Left a comment.

Resolved the comment and the issue. PTAL again.

**Review by sethiay — APPROVED** [review]

**Pull request merged**